topology.go

```diff
@@ -0,0 +1,216 @@
+}
+
+func (opts TopologyOptions) getConfigForComponents() string {
+	if !opts.IncludeConfig {
+		return emptyJSONBArraySQL
+	}
+	return `(
+        SELECT json_agg(json_build_object(
+            'id', config_items.id,
+            'name', config_items.name,
+            'external_type', config_items.external_type,
+            'config_type', config_items.config_type,
+            'cost_per_minute', config_items.cost_per_minute,
+            'cost_total_1d', config_items.cost_total_1d,
+            'cost_total_7d', config_items.cost_total_7d,
+            'cost_total_30d', config_items.cost_total_30d
+        )) FROM config_items
+        LEFT JOIN config_component_relationships ON config_items.id = config_component_relationships.config_id
+        WHERE config_component_relationships.component_id = topology_result.id AND config_component_relationships.deleted_at IS NULL
+        GROUP BY config_component_relationships.component_id
+    ) :: jsonb`
+}
+
+func (opts TopologyOptions) getConfigAnalysisForComponents() string {
+	if !opts.IncludeInsights {
+		return emptyJSONBArraySQL
+	}
+	return `(
+        SELECT json_agg(json_build_object(
+            'config_id', config_analysis.config_id,
+            'analyzer', config_analysis.analyzer,
+            'analysis_type', config_analysis.analysis_type,
+            'severity', config_analysis.severity
+        )) FROM config_analysis
+        LEFT JOIN config_component_relationships ON config_analysis.config_id = config_component_relationships.config_id
+        WHERE config_component_relationships.component_id = topology_result.id AND config_component_relationships.deleted_at IS NULL
+        GROUP BY config_component_relationships.component_id
+    ) :: jsonb`
+}
+
+func (opts TopologyOptions) getConfigAnalysisSummaryForComponents() string {
+	if !opts.IncludeInsightsSummary {
+		return emptyJSONBObjectSQL
+	}
+	return `(
+        SELECT json_object_agg(flatten.analysis_type, flatten.summary_json)
+        FROM (
+            SELECT summary.component_id, summary.analysis_type, json_object_agg(f.k, f.v) as summary_json
+            FROM (
+                SELECT config_component_relationships.component_id AS component_id, config_analysis.analysis_type, json_build_object(severity, count(*)) AS severity_agg
+                FROM config_analysis
+                LEFT JOIN config_component_relationships ON config_analysis.config_id = config_component_relationships.config_id
+                WHERE config_component_relationships.component_id = topology_result.id AND config_component_relationships.deleted_at IS NULL
+                GROUP BY config_analysis.severity, config_analysis.analysis_type, config_component_relationships.component_id
+            ) AS summary, json_each(summary.severity_agg) AS f(k,v) GROUP BY summary.analysis_type, summary.component_id
+        ) AS flatten GROUP BY flatten.component_id
+    ) :: jsonb`
+}
+
+func (p TopologyOptions) getIncidentsForComponents() string {
```

Comment:
I don't think we need this option - I can't think of a use case,  There is a reverse requirement though, get_incidents_by_component